examples/ava_tools/difficulty_from_csv.py

```diff
@@ -0,0 +1,128 @@
+    iw, ih = max(0.0, inter_x2 - inter_x1), max(0.0, inter_y2 - inter_y1)
+    inter = iw * ih
+    areaA = max(0.0, ax2-ax1) * max(0.0, ay2-ay1)
+    areaB = max(0.0, bx2-bx1) * max(0.0, by2-by1)
+    union = areaA + areaB - inter + 1e-8
+    return inter / union
+
+def load_ava_as(csv_path):
+    by_vid = defaultdict(list)
+    with open(csv_path, newline='') as f:
+        r = csv.reader(f)
+        for row in r:
+            vid, ts = row[0], float(row[1])
+            x1, y1, x2, y2 = map(float, row[2:6])
+            label = row[6].strip()
+            ent = row[7].strip()
+            by_vid[vid].append({
+                'ts': ts, 'box': (x1, y1, x2, y2),
+                'label': label, 'ent': ent
+            })
+    for vid in by_vid:
+        by_vid[vid].sort(key=lambda d: (d['ts'], d['ent']))
+    return by_vid
+
+def compute_difficulty(rows):
+    """
+    Compute difficulty metrics for a single video based on bounding boxes and labels.
+    Metrics:
+      - diversity: number of unique tracked entities (proxy for people)
+      - interactivity_avg / max: average and max faces per frame
+      - dynamics: average motion between frames (1 - IoU)
+      - occlusion: average overlap between faces
+      - audibility_sna_share: share of non-audible speaking labels
+    """
+    # group by timestamp
+    by_ts = defaultdict(list)
+    entities = set()
+    for r in rows:
+        by_ts[r['ts']].append(r)
+        entities.add(r['ent'])
+
+    # interactivity
+    faces_per_frame = [len(v) for v in by_ts.values()]
+    interactivity_avg = mean(faces_per_frame) if faces_per_frame else 0.0
+    interactivity_max = max(faces_per_frame) if faces_per_frame else 0
+
+    # diversity
+    diversity = len(entities)
+
+    # audibility: share of SPEAKING_NOT_AUDIBLE among speaking frames
+    speak = [r for r in rows if r['label'].startswith('SPEAKING')]
+    sna = [r for r in speak if 'NOT_AUDIBLE' in r['label']]
+    audibility_sna_share = (len(sna) / len(speak)) if speak else 0.0
+
+    # dynamics: mean (1 - IoU) per entity over successive frames
+    by_ent = defaultdict(list)
+    for r in rows:
+        by_ent[r['ent']].append(r)
+    dynamics_terms = []
+    for ent, seq in by_ent.items():
+        seq.sort(key=lambda d: d['ts'])
+        for i in range(1, len(seq)):
+            iou_val = iou(seq[i-1]['box'], seq[i]['box'])
+            dynamics_terms.append(1.0 - iou_val)
+    dynamics = mean(dynamics_terms) if dynamics_terms else 0.0
+
+    # occlusion: average pairwise IoU per frame when >=2 faces
+    occlusion_terms = []
+    for ts, dets in by_ts.items():
+        n = len(dets)
+        if n < 2:
+            continue
+        boxes = [d['box'] for d in dets]
+        # average over all pairs
+        s, c = 0.0, 0
+        for i in range(n):
+            for j in range(i+1, n):
+                s += iou(boxes[i], boxes[j])
+                c += 1
+        if c:
+            occlusion_terms.append(s / c)
```

Comment:
I do not get what c is doing - it is the number of combinations between detections if I'm correct. Why don't you just add all IOUs to the occlusion_terms? So you basically average for each frame and then later you average over all averages. Correct me if I'm wrong but I think that can be easier.